c/tskit/trees.c

```diff
@@ -5673,111 +5735,57 @@ tsk_tree_position_in_interval(const tsk_tree_t *self, double x)
     return self->interval.left <= x && x < self->interval.right;
 }
 
-/* NOTE:
- *
- * Notes from Kevin Thornton:
- *
- * This method inserts the edges for an arbitrary tree
- * in linear time and requires no additional memory.
- *
- * During design, the following alternatives were tested
- * (in a combination of rust + C):
- * 1. Indexing edge insertion/removal locations by tree.
- *    The indexing can be done in O(n) time, giving O(1)
- *    access to the first edge in a tree. We can then add
- *    edges to the tree in O(e) time, where e is the number
- *    of edges. This apparoach requires O(n) additional memory
- *    and is only marginally faster than the implementation below.
- * 2. Building an interval tree mapping edge id -> span.
- *    This approach adds a lot of complexity and wasn't any faster
- *    than the indexing described above.
- */
 static int
 tsk_tree_seek_from_null(tsk_tree_t *self, double x, tsk_flags_t TSK_UNUSED(options))
 {
     int ret = 0;
-    tsk_size_t edge;
-    tsk_id_t p, c, e, j, k, tree_index;
-    const double L = tsk_treeseq_get_sequence_length(self->tree_sequence);
-    const tsk_treeseq_t *treeseq = self->tree_sequence;
-    const tsk_table_collection_t *tables = treeseq->tables;
+    tsk_table_collection_t *tables = self->tree_sequence->tables;
     const tsk_id_t *restrict edge_parent = tables->edges.parent;
     const tsk_id_t *restrict edge_child = tables->edges.child;
-    const tsk_size_t num_edges = tables->edges.num_rows;
-    const tsk_size_t num_trees = self->tree_sequence->num_trees;
     const double *restrict edge_left = tables->edges.left;
     const double *restrict edge_right = tables->edges.right;
-    const double *restrict breakpoints = treeseq->breakpoints;
-    const tsk_id_t *restrict insertion = tables->indexes.edge_insertion_order;
-    const tsk_id_t *restrict removal = tables->indexes.edge_removal_order;
-
-    // NOTE: it may be better to get the
-    // index first and then ask if we are
-    // searching in the first or last 1/2
-    // of trees.
-    j = -1;
+    double interval_left = self->interval.left;
```

Comment:
You're not actually using these values that ``interval_left`` and ``interval_right`` are being initialised to. Better to just declare,
```c
double interval_left, interval_right;
```